src/sage/rings/polynomial/laurent_polynomial_ideal.py

```diff
@@ -399,17 +419,22 @@ def polynomial_ideal(self, saturate=True):
             Ideal (x + 3*y) of Multivariate Polynomial Ring in x, y
              over Rational Field
         """
-        if self._poly_ideal is not None and (self._saturated or not saturate):
-            return self._poly_ideal
```

Comment:
Why did you move this below?

Reply:
I think the goal was to simplify the generators in case it is a one variable Lairent ring (over a field, changed).

Reply:
Then you should add a test for the univariate case.